ompi/communicator/comm_cid.c

```diff
@@ -878,11 +898,14 @@ int ompi_comm_activate_nb (ompi_communicator_t **newcomm, ompi_communicator_t *c
         OMPI_COMM_SET_PML_ADDED(*newcomm);
     }
 
-    /* Step 1: the barrier, after which it is allowed to
-     * send messages over the new communicator
+    /**
+     * Dual-purpose barrier:
+     * 1. The communicator's disjointness is inferred fromimax_local_peers.
+     * 2. After the operation it is allowed to send messages over the new communicator.
+     *
      */
-    ret = context->allreduce_fn (&context->ok, &context->ok, 1, MPI_MIN, context,
-                                 &subreq);
+    ret = context->allreduce_fn (&context->max_local_peers, &context->max_local_peers, 1, MPI_MAX,
+                                 context, &subreq);
     if (OMPI_SUCCESS != ret) {
```

Comment:
Also, the check before was capable of identifying an error during communicator creation, and we need to maintain that capability.

Reply:
Huh.. could you explain more about this? I don't see `context->ok` being used for error checking - did I miss something?

Reply:
I'm still unsure about the magic of `context->ok` so I did not change this part. Please let me know otherwise.